### PR TITLE
Add truncation root for Native frames.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Native stacks that were truncated were being nested under the same [Truncated] frame for non-native frames. This adds a [Truncated] root for native truncated stacks.

### CODE REVIEW

File: packages/devtools/lib/src/timeline/cpu_profile_protocol.dart [hunk 1]
```diff
@@ -39,6 +39,11 @@ class CpuProfileData {
 
   void _processStackFrames(Response response) {
     final nativeRoot = CpuStackFrame('nativeRoot', '[Native]', 'Dart');
+    final nativeTruncatedRoot = CpuStackFrame(
+      'nativeTruncatedRoot',
```
Comment (jacob314) [inline packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: nit: looking at this code makes me think CpuStackFrame should take required named arguments. That would make it easier for me to tell what these 3 string arguments are when reviewing the code.

File: packages/devtools/lib/src/timeline/cpu_profile_protocol.dart [hunk 2]
```diff
@@ -49,13 +54,20 @@ class CpuProfileData {
       // TODO(kenzie): detect other native frames like "syscall" and "malloc"
       // once we get file paths in the stack frame json.
       if (stackFrameName.startsWith('[Native]')) {
-        parent ??= nativeRoot;
+        if (parent?.name == '[Truncated]') {
```
Comment (jacob314) [inline packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: nit: make '[Truncated]' a constant.

File: packages/devtools/lib/src/timeline/cpu_profile_protocol.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: done.

File: packages/devtools/lib/src/timeline/cpu_profile_protocol.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: done. Same for '[Native]'.

**Pull request merged**